File: AutomaticTimetableGeneratorSystem-main/timetable-frontend/src/services/timeSlotService.js
import api from "./api";

const BASE_URL = "/timeslots";

const timeSlotService = {

  getAllTimeSlots: async () => {

    const response = await api.get(BASE_URL);

    return response.data;

  },

  getTimeSlotById: async (id) => {

    const response = await api.get(`${BASE_URL}/${id}`);

    return response.data;

  },

  createTimeSlot: async (timeSlot) => {

    const response = await api.post(
      BASE_URL,
      timeSlot
    );

    return response.data;

  },

  updateTimeSlot: async (id, timeSlot) => {

    const response = await api.put(
      `${BASE_URL}/${id}`,
      timeSlot
    );

    return response.data;

  },

  deleteTimeSlot: async (id) => {

    const response = await api.delete(
      `${BASE_URL}/${id}`
    );

    return response.data;

  },

  getTimeSlotsByDay: async (dayOfWeek) => {

    const response = await api.get(
      `${BASE_URL}/day/${dayOfWeek}`
    );

    return response.data;

  },

  getTimeSlotsByOrder: async (slotOrder) => {

    const response = await api.get(
      `${BASE_URL}/order/${slotOrder}`
    );

    return response.data;

  },

};

export default timeSlotService;

File: AutomaticTimetableGeneratorSystem-main/timetable-backend/src/main/java/com/timetable/service/TimeSlotService.java
package com.timetable.service;

import java.util.List;

import com.timetable.dto.timeslot.TimeSlotRequest;
import com.timetable.dto.timeslot.TimeSlotResponse;

public interface TimeSlotService {

    /**
     * Save Time Slot
     */
    TimeSlotResponse saveTimeSlot(TimeSlotRequest request);

    /**
     * Update Time Slot
     */
    TimeSlotResponse updateTimeSlot(Long timeSlotId,
                                    TimeSlotRequest request);

    /**
     * Delete Time Slot
     */
    void deleteTimeSlot(Long timeSlotId);

    /**
     * Get Time Slot By ID
     */
    TimeSlotResponse getTimeSlotById(Long timeSlotId);

    /**
     * Get All Time Slots
     */
    List<TimeSlotResponse> getAllTimeSlots();

    /**
     * Get Time Slots By Day Of Week
     */
    List<TimeSlotResponse> getTimeSlotsByDayOfWeek(String dayOfWeek);

    /**
     * Get Time Slots By Slot Order
     */
    List<TimeSlotResponse> getTimeSlotsBySlotOrder(Integer slotOrder);

}